README.md

```diff
@@ -23,110 +23,128 @@ $ composer require tomphp/config-service-provider
 ```php
 <?php
 
-use League\Container\Container;
-use League\Container\ServiceProvider\AbstractServiceProvider;
-use TomPHP\ConfigServiceProvider\ConfigServiceProvider;
-
-class DatabaseConnectionProvider extends AbstractServiceProvider
-{
-    protected $provides = [
-        'database_connection',
-    ];
-    
-    public function register()
-    {
-        $this->container->share('database_connection', function () {
-            return new DatabaseConnection(
-                $this->container->get('config.db.name'),
-                $this->container->get('config.db.username'),
-                $this->container->get('config.db.password')
-            );
-        });
-    }
-}
+use League\Container\Container; // or Pimple\Container
+use TomPHP\ConfigServiceProvider\Configurator;
 
-$appConfig = [
+$config = [
     'db' => [
         'name'     => 'example_db',
         'username' => 'dbuser',
         'password' => 'dbpass',
-    ]
+    ],
+
+    'di' => [
+        'services' => [
+            'database_connection' => [
+                'class' => DatabaseConnection::class,
+                'arguments' => [
+                    'config.db.name',
+                    'config.db.username',
+                    'config.db.password',
+                ],
+            ],
+        ],
+    ],
 ];
 
 $container = new Container();
-
-$container->addServiceProvider(ConfigServiceProvider::fromConfig($appConfig));
-$container->addServiceProvider(new DatabaseConnectionProvider());
+Configurator::apply()->configFromArray($config)->to($container);
 
 $db = $container->get('database_connection');
 ```
 
-* Each item in the config array is added as a separate entry into the
-  container.
-* Each item name is has a prefix added to it. The prefix defaults to `config`.
-* If an item contains an sub-array, each of that array's are added separately
-  with a name made up of the first array key, followed by a separator (defaults
-  to `.`) followed by the key from the second array.
-
 ### Reading Files From Disk
 
-Instead of providing the config as an array, you can also provide a list of 
-file filesystem pattern matches to the `fromFiles` constructor.
+Instead of providing the config as an array, you can also provide a list of
+file pattern matches to the `fromFiles` function.
 
 ```php
-$container->addServiceProvider(ConfigServiceProvider::fromFiles([
-    'config_dir/*.global.php',
-    'json_dir/*.json',
-    'config_dir/*.local.php',
-]));
+Configurator::apply()
+     ->configFromFiles('config_dir/*.global.php')
+     ->configFromFiles('json_dir/*.json')
+     ->configFromFiles('config_dir/*.local.php')
+    ->to($container);
 ```
 
 #### Merging
 
-Patterns will be matched in the order the appear in the array. As files are
-read their config will be merged in, overwriting any matching keys.
+The reader matches files in the order the appear in the array.  As files are
```

Comment:
@tomphp 

Seems like we have an extra space here after the `.` - can we remove it?